src/app/tests/TestThreadBorderRouterManagementCluster.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/CommandHandler.h>
+#include <app/clusters/thread-border-router-management-server/thread-border-router-management-server.h>
+#include <cstdint>
+#include <cstring>
+#include <lib/core/CHIPError.h>
+#include <lib/support/BitFlags.h>
+#include <lib/support/Span.h>
+#include <lib/support/ThreadOperationalDataset.h>
+#include <lib/support/tests/ExtraPwTestMacros.h>
+#include <pw_unit_test/framework.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace GeneralCommissioning {
+// Mock function
+void SetBreadcrumb(Attributes::Breadcrumb::TypeInfo::Type breadcrumb) {}
+} // namespace GeneralCommissioning
+
+namespace ThreadBorderRouterManagement {
+// Mock functions
+bool IsFailSafeArmed(FabricIndex accessingFabricIndex)
+{
+    return true;
+}
+
+void DisarmFailSafeTimer() {}
+
+class TestDelegate : public Delegate
+{
+public:
+    TestDelegate()  = default;
+    ~TestDelegate() = default;
+
+    CHIP_ERROR Init() override { return CHIP_NO_ERROR; }
+
+    CHIP_ERROR GetPanChangeSupported(bool & panChangeSupported) override
+    {
+        panChangeSupported = mPanChangeSupported;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetBorderRouterName(MutableCharSpan & borderRouterName) override
+    {
+        size_t nameIndex = mUseInvalidBorderRouterName ? 1 : 0;
+        if (borderRouterName.size() > strlen(kTestName[nameIndex]))
+        {
+            memcpy(borderRouterName.data(), kTestName[nameIndex], strlen(kTestName[nameIndex]));
```

Comment:
This is a buffer overrun if things are done properly. It should never be allowed that we hit the "too long". The API should be constructed that the "exact" max size is used in all APIs, and then `CopyCharSpanToMutableCharSpan` should be used. String that don't fit MUST be truncated, rather than attempting to detect overflow.

Reply:
Done